core/src/main/java/software/amazon/awssdk/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -199,10 +199,10 @@ private void beforeExecute() {
 
         private CompletableFuture<Response<OutputT>> doExecute() throws Exception {
             if (isRetry()) {
-                resetRequestInputStream(request.getContent());
+                resetRequestInputStream(request.content().orElse(null));
```

Comment:
Can you do an ifPresent so you don't have to do a null check in the method?

Reply:
+1